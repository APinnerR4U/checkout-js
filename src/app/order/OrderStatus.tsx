import { Order } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent } from 'react';

import { TranslatedHtml, TranslatedString } from '../locale';

import OrderConfirmationSection from './OrderConfirmationSection';

export interface OrderStatusProps {
    supportEmail?: string;
    supportPhoneNumber?: string;
    order: Order;
}

const OrderStatus: FunctionComponent<OrderStatusProps> = ({
    order,
    supportEmail,
    supportPhoneNumber,
}) => {
    const isPendingReview = order.status === 'MANUAL_VERIFICATION_REQUIRED';
    const orderNumber = order.orderId;

    return <OrderConfirmationSection>
        { orderNumber &&
        <p data-test="order-confirmation-order-number-text">
            <TranslatedHtml
                id="order_confirmation.order_number_text"
                data={ { orderNumber } }
            />
        </p> }

        { isPendingReview &&
        <p>
            <TranslatedString
                id="order_confirmation.order_pending_review_text"
            />
        </p> }

        { !isPendingReview &&
        <p>
            <TranslatedHtml
                id={ supportPhoneNumber ?
                    'order_confirmation.order_with_support_number_text' :
                    'order_confirmation.order_without_support_number_text' }
                data={ { orderNumber, supportEmail, supportPhoneNumber } }
            />
        </p> }

        { order.hasDigitalItems &&
        <p>
            <TranslatedHtml
                id={ order.isDownloadable ?
                    'order_confirmation.order_with_downloadable_digital_items_text' :
                    'order_confirmation.order_without_downloadable_digital_items_text' }
            />
        </p> }
    </OrderConfirmationSection>;
};

export default OrderStatus;
