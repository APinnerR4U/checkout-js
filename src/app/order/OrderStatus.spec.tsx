import { Order } from '@bigcommerce/checkout-sdk';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { getOrder } from './orders.mock';
import OrderStatus from './OrderStatus';

let order: Order;
let orderStatus: ShallowWrapper;

describe('OrderStatus', () => {
    describe('when order is not waiting manual verification', () => {

        beforeEach(() => {
            order = getOrder();
        });

        describe('and has support data', () => {
            beforeEach(() => {
                orderStatus = shallow(<OrderStatus
                    order={ order }
                    supportPhoneNumber="990"
                    supportEmail="test@example.com"
                />);
            });

            it('renders status with contact information', () => {
                expect(orderStatus).toMatchSnapshot();
            });
        });

        describe('and has no support data', () => {
            beforeEach(() => {
                orderStatus = shallow(<OrderStatus
                    order={ order }
                />);
            });

            it('renders status', () => {
                expect(orderStatus).toMatchSnapshot();
            });
        });
    });

    describe('when order is waiting manual verification', () => {
        beforeEach(() => {
            orderStatus = shallow(<OrderStatus
                order={ {
                    ...getOrder(),
                    status: 'MANUAL_VERIFICATION_REQUIRED',
                } }
            />);
        });

        it('renders status with special text', () => {
            expect(orderStatus).toMatchSnapshot();
        });
    });

    describe('when order has digital items', () => {
        beforeEach(() => {
            order = {
                ...getOrder(),
                hasDigitalItems: true,
            };
        });

        describe('and has downloadable items', () => {
            beforeEach(() => {
                order.isDownloadable = true;
            });

            it('renders status with downloadable items text', () => {
                expect(shallow(<OrderStatus order={ order }/>)).toMatchSnapshot();
            });
        });

        describe('and does not have downloadable items', () => {
            beforeEach(() => {
                order.isDownloadable = false;
            });

            it('renders status without downloadable items text', () => {
                expect(shallow(<OrderStatus order={ order }/>)).toMatchSnapshot();
            });
        });
    });
});
