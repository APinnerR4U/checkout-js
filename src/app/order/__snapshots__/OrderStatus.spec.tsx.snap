// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderStatus when order has digital items and does not have downloadable items renders status without downloadable items text 1`] = `
<OrderConfirmationSection>
  <p
    data-test="order-confirmation-order-number-text"
  >
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
        }
      }
      id="order_confirmation.order_number_text"
    />
  </p>
  <p>
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
          "supportEmail": undefined,
          "supportPhoneNumber": undefined,
        }
      }
      id="order_confirmation.order_without_support_number_text"
    />
  </p>
  <p>
    <WithLanguage(TranslatedHtml)
      id="order_confirmation.order_without_downloadable_digital_items_text"
    />
  </p>
</OrderConfirmationSection>
`;

exports[`OrderStatus when order has digital items and has downloadable items renders status with downloadable items text 1`] = `
<OrderConfirmationSection>
  <p
    data-test="order-confirmation-order-number-text"
  >
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
        }
      }
      id="order_confirmation.order_number_text"
    />
  </p>
  <p>
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
          "supportEmail": undefined,
          "supportPhoneNumber": undefined,
        }
      }
      id="order_confirmation.order_without_support_number_text"
    />
  </p>
  <p>
    <WithLanguage(TranslatedHtml)
      id="order_confirmation.order_with_downloadable_digital_items_text"
    />
  </p>
</OrderConfirmationSection>
`;

exports[`OrderStatus when order is not waiting manual verification and has no support data renders status 1`] = `
<OrderConfirmationSection>
  <p
    data-test="order-confirmation-order-number-text"
  >
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
        }
      }
      id="order_confirmation.order_number_text"
    />
  </p>
  <p>
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
          "supportEmail": undefined,
          "supportPhoneNumber": undefined,
        }
      }
      id="order_confirmation.order_without_support_number_text"
    />
  </p>
</OrderConfirmationSection>
`;

exports[`OrderStatus when order is not waiting manual verification and has support data renders status with contact information 1`] = `
<OrderConfirmationSection>
  <p
    data-test="order-confirmation-order-number-text"
  >
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
        }
      }
      id="order_confirmation.order_number_text"
    />
  </p>
  <p>
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
          "supportEmail": "test@example.com",
          "supportPhoneNumber": "990",
        }
      }
      id="order_confirmation.order_with_support_number_text"
    />
  </p>
</OrderConfirmationSection>
`;

exports[`OrderStatus when order is waiting manual verification renders status with special text 1`] = `
<OrderConfirmationSection>
  <p
    data-test="order-confirmation-order-number-text"
  >
    <WithLanguage(TranslatedHtml)
      data={
        Object {
          "orderNumber": 295,
        }
      }
      id="order_confirmation.order_number_text"
    />
  </p>
  <p>
    <WithLanguage(TranslatedString)
      id="order_confirmation.order_pending_review_text"
    />
  </p>
</OrderConfirmationSection>
`;
